spec/lib/advanced_search_builder_spec.rb

```diff
@@ -57,4 +57,47 @@ def initialize(blacklight_config)
       end
     end
   end
+
+  describe "#add_advanced_search_to_solr" do
+    let(:to_solr) { { q: 'advanced query', fq: 'inclusive facet' } }
+    let(:advanced_query) { double('advanced_query') }
+    let(:solr_params) { { q: 'basic', fq: 'original fq' } }
+    before do
+      allow(advanced_query).to receive(:to_solr).and_return(to_solr)
```

Comment:
the q parameter of the blacklight_params doesn't affect whether the solr query gets rewritten or not so maybe I should just remove it? The 'advanced query' should only replace the original `q: 'basic'` when the blacklight_params search_field matches the advanced search url_key